Allow two icons on iconcard

HeleneKassandra opened the pull request:
<!-- Gi saken en oppsummerende tittel ovenfor. -->

## Beskrivelse
Legger til muligheten til å ha to ikoner samtidig på iconCard. 
![image](https://github.com/user-attachments/assets/1097bef1-3c26-45c2-9d78-4f42cac225b5)

Denne endringen lager en breaking change, da jeg fjerner `iconPosition`-prop som tidligere ble lagt til for at man kunne ha ikon på høyre-side og ikke bare venstre. Ved å tillatte 2 ikoner, så kan denne problemstillingen løses ved at man kun sender inn `rightIcon`-ikon

<!-- Detaljert beskrivelse av endringene dine. Skjermskudd er lov. -->

## Motivasjon og kontekst
Vi har flere steder i Oversikt der vi bruker iconCard, men har 2 ikoner. 
Dette er også allerede tillatt i StippledCard, så gjør ett forsøk på å legge det til i iconCard også. 
<!-- Hvorfor trengs denne endringen, og hva løser den? -->

## Testing
Kjørt opp lokalt og sjekket i Storybook. 
<!-- Hvordan har du har testet endringene dine? Ta gjerne med systeminfo o.l -->

<!--

Helt til slutt, har du ..

.. lest retningslinjene våre for bidrag?
.. tatt en siste sjekk for kode og skrivefeil, debug informasjon o.l?
.. kjørt og eventuelt oppdatert testene?
.. kommentert og dokumentert det som trengs?
.. knyttet denne opp til relaterte issues?

-->

HeleneKassandra committed:
fix(.storybook): add stories

HeleneKassandra commented:
For å få minst mulig endringer så beholdt jeg måten det ble rendret på i dag: 
```
{iconElement}
{bodyElement}
{rightIconElement}
```

men her kunne man kanskje bare inlinet det?

dagfrode commented on packages/ffe-cards-react/src/IconCard/IconCard.tsx:
```diff
@@ -13,16 +13,15 @@ export type IconCardProps<As extends ElementType = 'div'> = Omit<
 > & {
     /** Element of icon */
-    icon: ReactElement;
+    icon?: ReactElement;
```
Hadde det gitt mening med `leftIcon` fremfor å beholde `icon`?

HeleneKassandra replied:
Ja, men jeg lot være for å unngå enda flere breaking changes. Per nå så er det egentlig kun de som bruker "iconPosition" som blir rammet av breaking change.  Dette er vel også mønsteret i StippledCard per nå.

dagfrode replied:
Fair, blir nok en gjennomgang av cards etterhvert så Figma og koden blir likere :sweat_smile:

HeleneKassandra replied:
Er nok smart :) Ta gjerne kontakt om dere ønsker sparring på kort, vi har veldig mange forskjellige bruksområder og varianter hos oss i Team Oversikt, som jeg ville elsket å få inn i Designsystemet.

dagfrode commented on packages/ffe-cards-react/src/IconCard/IconCard.spec.tsx:
```diff
@@ -61,26 +61,20 @@ describe('IconCard', () => {
         const card = screen.getByRole('listitem');
-        const body = card.querySelector('.ffe-icon-card__body') as Element;
-        const icon = card.querySelector('.ffe-icon-card__icon') as Element;
+        const icons = card.querySelectorAll('.ffe-icon-card__icon') as NodeListOf<Element>;
```
`as NodeListOf<Element>` feiler da den ikke finner typen `NodeListOf`, men den ser ut til å være redundant anyways :)

HeleneKassandra replied:
Jeg fikk error da det bare var Element 😓

dagfrode replied:
Huh, merkelig

dagfrode committed:
feat(ffe-cards): allow two icons on iconCard

BREAKING CHANGE: removes "iconPosition"-related styling

dagfrode committed:
feat(ffe-cards-react): allow two icons on iconCard

BREAKING CHANGE: removes and replaces "IconPosition"-prop and styling with "rightIcon" prop.
